src/modules/accommodations/accommodations.controller.ts

```diff
@@ -23,4 +43,36 @@ export class AccommodationsController {
       limit: searchQueryParams.limit,
     };
   }
+
+  @Post()
+  @CreateAccommodationSwaggerDecorator()
+  async createAccommodation(@Body() createDto: CreateAccommodationDto) {
+    return this.accommodationsService.createAccommodation(createDto);
+  }
+
+  @Get(':id')
+  @GetByIdSwaggerDecorator()
+  async getAccommodationById(@Param('id') id: string) {
+    return this.accommodationsService.getAccommodationById(id);
+  }
+
+  @Put(':id')
+  @UpdateAccommodationSwaggerDecorator()
+  async updateAccommodation(
+    @Param('id') id: string,
+    @Body() updateDto: UpdateAccommodationDto,
+  ) {
+    return this.accommodationsService.updateAccommodation(id, updateDto);
+  }
+
+  @Delete(':id')
+  @DeleteAccommodationSwaggerDecorator()
+  async deleteAccommodation(@Param('id') id: string) {
```

Comment:
you are not checking if the id is uuid here. use the decorator that I created "@ParseUUIDV4Pipe"